src/Compilers/CSharp/Portable/Symbols/Source/SourceConstructorSymbol.cs

```diff
@@ -192,11 +192,26 @@ private void CheckModifiers(MethodKind methodKind, bool hasBody, Location locati
             }
         }
 
+#nullable enable
         internal override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarations()
         {
+            // Attributes on partial constructors are owned by the definition part.
+            // If this symbol has a non-null PartialDefinitionPart, we should have accessed this method through that definition symbol instead.
+            Debug.Assert(PartialDefinitionPart is null);
+
+            if (SourcePartialImplementationPart is { } implementationPart)
+            {
+                return OneOrMany.Create(
+                    ((ConstructorDeclarationSyntax)this.SyntaxNode).AttributeLists,
```

Comment:
It looks like for `SourceOrdinaryMethodSymbol.AttributeDeclarationSyntaxList`, we actually check `SourceMemberContainerSymbol.AnyMemberHasAttributes` on the `ContainingType`, before accessing the `AttributeLists`. I believe this avoids creating green nodes in the common case of no attributes.

It turns out we were not consistent about this with properties and accessors. If you wish, you could try to improve this, by gating this on the same check. If, indeed, it actually turns out to be an improvement at all.